app/Application/Services/WalletCryptocurrenciesService.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Application\Services;
+
+use App\Application\Exceptions\WalletNotFoundException;
+use App\Domain\DataSources\WalletDataSource;
+use Illuminate\Support\Facades\Cache;
+
+class WalletCryptocurrenciesService
+{
+    private WalletDataSource $walletDataSource;
+
+    public function __construct(WalletDataSource $walletDataSource)
+    {
+        $this->walletDataSource = $walletDataSource;
+    }
+
+    /**
+     * @throws WalletNotFoundException
+     */
+    public function getWalletCryptocurrencies($wallet_id): array
+    {
+        $walletId = $this->walletDataSource->findById($wallet_id);
+        if ($walletId === null) {
+            throw new WalletNotFoundException();
+        }
+
+        $wallet = Cache::get('wallet_' . $wallet_id);
```

Comment:
This must be here??

Reply:
```suggestion
        $wallet = $this->walletDataSource->getWalletById($wallet_id);
```
What do you think of create this function?